src/worker/move-contact-worker.ts

```diff
@@ -40,40 +46,56 @@ export class MoveContactWorker {
     const jobData: MoveContactData = job.data;
 
     // Ensure server availability
-    if (await this.shouldPostpone(jobData)) {
-      await this.postpone(job, processingToken);
+    const { shouldPostpone, reason } = await this.shouldPostpone(jobData);
+    if (shouldPostpone) {
+      await this.postpone(job, reason, processingToken);
       throw new DelayedError();
     }
 
-    const result = await this.moveContact(jobData);
+    const result = await this.moveContact(job);
     if (!result.success) {
-      job.log(`[${new Date().toISOString()}]: ${result.message}`);
       const errorMessage = `Job ${job.id} failed with the following error: ${result.message}`;
       console.error(errorMessage);
+      this.jobLogWithTimestamp(job, errorMessage);
       throw new Error(errorMessage);
     }
 
     console.log(`Job completed successfully: ${job.id}`);
     return true;
   };
 
-  private static async shouldPostpone(jobData: MoveContactData): Promise<boolean> {
+  private static handleRetryBackoff = (
+    attemptsMade: number, type: string | undefined, err: Error | undefined, job: MinimalJob | undefined
+  ): number => {
+    const {retryTimeFormatted} = this.computeRetryTime();
+    const fullMessage = `Job ${job?.id} will be retried ${attemptsMade + 1} time at ${retryTimeFormatted}. Due to failure: ${type}: ${err?.message}`;
+    this.jobLogWithTimestamp(job, fullMessage);
+    return this.DELAY_IN_MILLIS;
+  };
+
+  private static async shouldPostpone(jobData: MoveContactData): Promise<{ shouldPostpone: boolean; reason: string }> {
     try {
       const { instanceUrl } = jobData;
       const response = await axios.get(`${instanceUrl}/api/v2/monitoring`);
       const sentinelBacklog = response.data.sentinel?.backlog;
       console.log(`Sentinel backlog at ${sentinelBacklog} of ${this.MAX_SENTINEL_BACKLOG}`);
-      return sentinelBacklog > this.MAX_SENTINEL_BACKLOG;
+      return { shouldPostpone: sentinelBacklog > this.MAX_SENTINEL_BACKLOG, reason: `Sentinel backlog too high at ${sentinelBacklog}` };
     } catch (err: any) {
       const errorMessage = err.response?.data?.error?.message || err.response?.error || err?.message;
       console.error('Error fetching monitoring data:', errorMessage);
-      return true;
+
+      // Handle server unavailability (HTTP 500 errors)
+      if (err.response?.status === 500) {
+        console.log('Server error encountered, postponing job...');
+        return { shouldPostpone: true, reason: `Server error encountered: ${errorMessage}` };
+      }
+      return { shouldPostpone: false, reason: '' };
```

Comment:
This seems a bit odd here. Would it be clearer if this function returned `Promise<PostponeReason>` (or alike) and returned undefined when postponing is not required?